### PR TITLE
Add tests, first for Ormolu formatter

### PR DESCRIPTION
**Pull request by alanz**
Its a mess, but this is the first pass.

### CODE REVIEW

File: test/testdata/stack.yaml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# WARNING: THIS FILE IS AUTOGENERATED IN test/utils/TestUtils. IT WILL BE OVERWRITTEN ON EVERY TEST RUN
+resolver: lts-14.22
+packages:
+- '.'
+extra-deps: []
+flags: {}
+extra-package-dbs: []
```
Comment (fendor) [inline test/testdata/stack.yaml]: Should not be committed afaik?

File: test/exe/Main.hs [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+-}
+
+mkRange :: Int -> Int -> Int -> Int -> Range
+mkRange a b c d = Range (Position a b) (Position c d)
+
+run :: Session a -> IO a
+run s = withTempDir $ \dir -> do
+  let ghcideExe = hieCommand
+
+  -- Temporarily hack around https://github.com/mpickering/hie-bios/pull/56
```
Comment (fendor) [inline test/exe/Main.hs]: Do we still have to work around this? The pr is closed.

Comment (alanz) [reply test/exe/Main.hs]: I just copied this file as is from hie. If it is gone there, it can go here.

Comment (fendor) [reply test/exe/Main.hs]: Isnt that from ghcide?

Comment (alanz) [reply test/exe/Main.hs]: True. And unused, in favour of the the hspec stuff. I guess I can just delete that whole directory.

**Pull request merged**